rsocket-core/src/main/java/io/rsocket/core/RSocketConnector.java

```diff
@@ -468,8 +469,19 @@ public RSocketConnector payloadDecoder(PayloadDecoder decoder) {
    * @return a {@code Mono} with the connected RSocket
    */
   public Mono<RSocket> connect(Supplier<ClientTransport> transportSupplier) {
+
     Mono<DuplexConnection> connectionMono =
-        Mono.fromSupplier(transportSupplier).flatMap(t -> t.connect(mtu));
+        Mono.fromSupplier(transportSupplier)
+            .flatMap(
+                transport -> {
+                  Mono<DuplexConnection> errorMono = FragmentationDuplexConnection.checkMtu(mtu);
```

Comment:
This is redundant since we have already checked `mtu` validity on the factory level